sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -922,6 +922,22 @@ object SQLConf {
     .checkValues(Set("none", "uncompressed", "snappy", "gzip", "lzo", "lz4", "brotli", "zstd"))
     .createWithDefault("snappy")
 
+  val PARQUET_COMPRESSION_ZSTD_LEVEL = buildConf("spark.sql.parquet.zstd.level")
```

Comment:
what about the existing config `spark.io.compression.zstd.level`? Can't we just reuse the existing config instead of introducing new ones for each file format?

Reply:
> what about the existing config `spark.io.compression.zstd.level`? Can't we just reuse the existing config instead of introducing new ones for each file format?

spark.io.compression.zstd.level is adapted to `spark.io.compression.codec`. It only works on internal data, and does not use the `parquet-mr` library, that is, it is used in a different way. I don't think it can be replicated use.

<img width="908" alt="image" src="https://user-images.githubusercontent.com/13965087/180647006-08978254-08c6-41ac-be91-40b02caaeff5.png">